web/libs/editor/src/tags/control/Number.jsx

```diff
@@ -171,15 +172,15 @@ const Model = types
       },
 
       increaseValue() {
-        if (self.number >= Number(self.max)) {
-          self.setNumber(0);
-        } else {
-          if (self.number > 0) {
-            self.setNumber(self.number + 1);
-          } else {
-            self.setNumber(1);
-          }
-        }
+        const min = safeNumber(self.min, 0);
+        const max = safeNumber(self.max, Number.POSITIVE_INFINITY);
+        const step = positiveNumber(self.step);
+        const defaultValue = safeNumber(self.defaultvalue, min);
+        const current = safeNumber(self.number, defaultValue);
+
+        const next = current + step > max ? min : current + step;
```

Comment:
should not we hit `max` value and stop there and not falling back to `min`?
```suggestion
        const next = current + step > max ? max : current + step;
```

Reply:
The current flow is to loop, so I replicate the action. I agree with the current loop behavior because it always provides feedback.
The current bug looks like a bug because the input stays at 0.
Tell me if you feel strong here.

@hlomzik  thanks so much for go deep in the review, awesome!

Reply:
Btw in this one I am 50% so I will apply your take. Since will happen after several iterations

Reply:
We discussed offline and we both are 50% on one direction over another, so we'll keep it as is for now.